packages/web-frontend/src/historyNavigation/utils.js

```diff
@@ -87,6 +89,29 @@ export const isLocationEqual = (a, b) => {
   return true;
 };
 
+export const convertUrlPeriodStringToObject = (periodString, periodGranularity) => {
+  const [startDate, endDate] = periodString.split('-');
+
+  return {
+    startDate: moment(startDate, GRANULARITY_CONFIG[periodGranularity].urlFormat),
+    endDate: moment(endDate || startDate, GRANULARITY_CONFIG[periodGranularity].urlFormat),
+  };
+};
+
+export const convertObjectToUrlPeriodString = ({ startDate, endDate }, periodGranularity) => {
+  if (GRANULARITIES_WITH_ONE_DATE.includes(periodGranularity)) {
+    if (!startDate.isSame(endDate)) {
+      console.error(
```

Comment:
How should we handle this case?

Reply:
As discussed, while this may indicate an error (start and end date being different for a one-date granularity), it's not our concern here. So we could as well as remove the check and error logging, and leave this concern to places where this state of data does matter